token/cli/src/main.rs

```diff
@@ -2916,6 +3028,97 @@ fn app<'a, 'b>(
                     .help("Specify the authority keypair. Defaults to the client keypair address.")
                 )
         )
+        .subcommand(
+            SubCommand::with_name(CommandName::InitializeMetadata.into())
+                .about("Initialize metadata extension on a token mint")
+                .arg(
+                    Arg::with_name("token")
+                        .validator(is_valid_pubkey)
+                        .value_name("TOKEN_MINT_ADDRESS")
+                        .takes_value(true)
+                        .required(true)
+                        .index(1)
+                        .help("The token address with no metadata present"),
+                )
+                .arg(
+                    Arg::with_name("name")
+                        .value_name("TOKEN_NAME")
+                        .takes_value(true)
+                        .index(2)
+                        .help("The name of the token to set in metadata"),
+                )
+                .arg(
+                    Arg::with_name("symbol")
+                        .value_name("TOKEN_SYMBOL")
+                        .takes_value(true)
+                        .index(3)
+                        .help("The symbol of the token to set in metadata"),
+                )
+                .arg(
+                    Arg::with_name("uri")
+                        .value_name("TOKEN_URI")
+                        .takes_value(true)
+                        .index(4)
+                        .help("The URI of the token to set in metadata"),
+                )
+                .arg(
+                    Arg::with_name("mint_authority")
+                        .long("mint-authority")
+                        .alias("owner")
+                        .value_name("KEYPAIR")
+                        .validator(is_valid_signer)
+                        .takes_value(true)
+                        .help(
+                            "Specify the mint authority keypair. \
+                             This may be a keypair file or the ASK keyword. \
+                             Defaults to the client keypair."
+                        ),
+                )
```

Comment:
re: my other comment: update authority is missing from this list of args

Reply:
Ah yeah, good point. Sorry, I was doing this by default during initialize before, and then it didn't work, so I created this other instruction. I'll definitely add it in